crates/sui-config/src/dynamic_transaction_signing_checks.rs

```diff
@@ -0,0 +1,262 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{
+    de::{self, Deserializer},
+    ser::Serializer,
+    Deserialize,
+};
+use serde_json::Value as JsonValue;
+use starlark::{
+    environment::{Globals, Module},
+    eval::Evaluator,
+    syntax::{AstModule, Dialect, DialectTypes},
+    values::{dict::AllocDict, AllocValue, Heap, Value},
+    ErrorKind,
+};
+use sui_types::{
+    base_types::ObjectRef,
+    signature::GenericSignature,
+    transaction::{InputObjectKind, TransactionData},
+};
+use tracing::warn;
+
+/// The name of the input variables that the transaction will have in the Starlark program.
+const TX_DATA_NAME: &str = "tx_data";
+const TX_SIGNERS_NAME: &str = "tx_signers";
+const TX_INPUT_OBJECTS_NAME: &str = "tx_input_objects";
+const TX_RECEIVING_OBJECTS_NAME: &str = "tx_receiving_objects";
+
+/// The dummy name of the Starlark file being executed. We will just be passing the string of the
+/// program in directly so this is not important but may appear in error messages.
+const STAR_INPUT_FILE_NAME: &str = "dummy.star";
+
+#[derive(Debug, thiserror::Error)]
+pub enum DynamicCheckRunnerError {
+    #[error("Failed to serialize transaction data to JSON: {0}")]
+    JSONSerializationError(String),
+    #[error("Failed to parse Starlark program value -- unsupported number type {0}")]
+    UnsupportedNumberFormat(String),
+    #[error(
+        "Failed to execute Starlark program -- invalid return type expected a bool but got {0}"
+    )]
+    InvalidReturnType(String),
+    #[error("Failed to execute Starlark program: {0}")]
+    ExecutionError(ErrorKind),
+    #[error("Failed to load Starlark program: {0}")]
+    LoadingError(ErrorKind),
+    #[error("Check failed -- transaction denied")]
+    CheckFailure,
+}
+
+#[derive(Debug, Clone)]
+pub struct DynamicCheckRunnerContext {
+    module: AstModule,
+    globals: Globals,
+    loaded_program: String,
+}
+
+const DIALECT: Dialect = Dialect {
+    enable_def: true,
+    enable_lambda: true,
+    enable_keyword_only_arguments: false,
+    enable_positional_only_arguments: false,
+    enable_types: DialectTypes::Disable,
+    // NB: set loader to false to prevent any external loading
+    enable_load: false,
+    enable_load_reexport: false,
+    // NB: Allow for top level statements to be used (e.g., top-level `for`, `if`, etc.)
+    enable_top_level_stmt: true,
+    enable_f_strings: false,
+    _non_exhaustive: (),
```

Comment:
I think this is supposed to end with a `..Dialect::Standard` instead of the `_non_exhaustive` being stamped in here -- otherwise it defeats the point of having a `_non_exhaustive` field (this will just fail to compile once a new field is added).

Maybe we want that so we don't accidentally gain a new configuration we didn't want -- but if so, a comment would be helpful.

Reply:
Yea, I think here we definitely want to break if a new field is added. Will add a comment!